plugin/dnstap/writer.go

```diff
@@ -13,6 +15,7 @@ import (
 type ResponseWriter struct {
 	queryTime time.Time
 	query     *dns.Msg
+	ctx       *context.Context
```

Comment:
do you really need a pointer here?

Reply:
The ctx field is needed in the WriteMsg function below to call TapMessageWithMetadata.

Or do you mean to use the context struct directly without using a pointer? Like ctx context.Context? I think using a pointer can help to avoid an extra copy of the context structure, thus reducing memory usage.

Reply:
Yes, I mean storing context struct directly. I don't think this is critical memory overhead. This is like storing/passing pointers to strings everywhere instead of storing/passing strings.
Also, pointers are managed by garbage collector subsystem which may spend some extra memory/CPU on serving pointers.